src/celeritas/ext/GeantOpticalPhysicsOptions.hh

```diff
@@ -0,0 +1,135 @@
+//---------------------------------------------------------------------------//
+//! \file celeritas/ext/GeantOpticalPhysicsOptions.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Types.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+//! WLS time model selection
+enum class WLSTimeProfileSelection
+{
+    none,
+    delta,  //!< Delta function
+    exponential,  //!< Exponential decay
+    size_
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Construction options for Geant optical physics.
+ *
+ * These options attempt to default to our closest match to \c
+ * G4OpticalPhysics from Geant4 10.5 onwards.
+ */
+struct GeantOpticalPhysicsOptions
+{
+    //!@{
+    //! \name Optical photon physics
+    //!@}
+
+    //!@{
+    //! \name Optical photon creation physics
+    //! Enable Cerenkov radiation
+    bool cerenkov_radiation{true};
+    //! Enable scintillation
+    bool scintillation{true};
+    //!@}
+
+    //!@{
+    //! \name Optical photon physics
+    //! Enable wavelength shifting and select a time profile
+    WLSTimeProfileSelection wavelength_shifting{WLSTimeProfileSelection::delta};
+    //! Enable second wavelength shifting type and select a time profile (TODO:
+    //! clarify)
+    WLSTimeProfileSelection wavelength_shifting2{
+        WLSTimeProfileSelection::delta};
+    //! Enable boundary effects
+    bool boundary{true};
+    //! Enable absorption
+    bool absorption{true};
+    //! Enable Rayleigh scattering
+    bool rayleigh_scattering{true};
+    //! Enable Mie scattering
+    bool mie_scattering{true};
+    //!@}
+
+    //!@{
+    //! \name Cerenkov physics options
+    //! Enable generation of Cerenkov photons
+    bool cerenkov_stack_photons{true};
+    //! Track generated photons before parent
+    bool cerenkov_track_secondaries_first{true};
+    //! Maximum number of photons that can be generated before limiting step
+    int cerenkov_max_photons{100};
+    //! Maximum percentage change in particle \f$\beta\f$  before limiting step
+    double cerenkov_max_beta_change{10.0};
+    //!@}
+
+    //!@{
+    //! \name Scintillation physics options
+    //! Enable generation of scintillation photons
+    bool scint_stack_photons{true};
+    //! Track generated photons before parent
+    bool scint_track_secondaries_first{true};
+    //! Use per-particle yield and time constants for photon generation
+    bool scint_by_particle_type{false};
+    //! Use material properties for sampling photon generation time
+    bool scint_finite_rise_time{false};
+    //! Attach scintillation interaction information to generated photon
+    bool scint_track_info{false};
+    //!@}
+
+    //!@{
+    //! \name Boundary physics options
+    //! Invoke Geant4 SD at post step point if photon deposits energy
+    bool invoke_sd{false};
+    //!@}
+
```

Comment:
That makes sense, though perhaps we leave for now as a reminder (I can comment out) until there's a mechanism to do this in Celeritas? I'd guess it'll eventually go in `SDSetupOptions` or similar.

Reply:
O.k., latest commit implements the `operator bool` retaining the previous behaviour, so

- default construction enables all Optical processes
- The static function is "deactivated" rather than "all_physics"

So opposite to what you suggested @sethrj, but can easily flip this. It kind of felt more natural for default construction to enable everything, but I don't have a strong preference.